### PR TITLE
Use byte for colorquad

### PR DESCRIPTION
**Pull request by JasperGeurtz**
https://github.com/JavaBWAPI/JBWAPI/pull/53#issuecomment-583028008

### CODE REVIEW

File: src/main/java/bwapi/Color.java [hunk 1]
```diff
@@ -173,24 +173,24 @@ private static int getRGBIndex(final int red, final int green, final int blue) {
...
     }
 
     public int red() {
-        return id < 256 ? defaultPalette[id].rgbRed : 0;
+        return id < 256 ? defaultPalette[id].rgbRed & 0xFF : 0;
     }
 
     public int green() {
-        return id < 256 ? defaultPalette[id].rgbGreen : 0;
+        return id < 256 ? defaultPalette[id].rgbGreen & 0xFF : 0;
```
Comment (Bytekeeper) [inline src/main/java/bwapi/Color.java]: Although not part of this change, we could move the `id < 256` to the constructor - all other places seem to result in either 0 or 255 and will always be `< 256`

**Review by dgant — APPROVED** [review]

**Review by Bytekeeper — APPROVED** [review]

File: src/main/java/bwapi/Color.java (hunk 1 above)
Comment (JasperGeurtz) [reply src/main/java/bwapi/Color.java]: Within the JBWAPI yes, but theoretically the BWAPI Server could deliver a non 0-255 sized in for example:
https://github.com/JavaBWAPI/JBWAPI/blob/95b640d5d4f4d2fb49ee62b944f369e9c520b52b/src/main/java/bwapi/Player.java#L536

However in practice this should never be possible, as Color is a Type with a max of 255, so everything (unsigned) above it, will be set to 255:
https://github.com/bwapi/bwapi/blob/3438abd8e0222f37934ba62b2130c3933b067678/bwapi/include/BWAPI/Type.h#L66

So I'll take your suggestion and put it in the constructor
